src/Microsoft.AspNetCore.Mvc.Testing/MvcWebApplicationBuilder.cs

```diff
@@ -0,0 +1,137 @@
+        }
+
+        /// <summary>
+        /// Configures <see cref="ApplicationPartManager"/> to include the default set
+        /// of <see cref="ApplicationPart"/> provided by <see cref="DefaultAssemblyPartDiscoveryProvider"/>.
+        /// </summary>
+        /// <returns>An instance of this <see cref="MvcWebApplicationBuilder{TStartup}"/></returns>
+        public MvcWebApplicationBuilder<TStartup> UseApplicationAssemblies()
+        {
+            ApplicationAssemblies.AddRange(DefaultAssemblyPartDiscoveryProvider
+                .DiscoverAssemblyParts(typeof(TStartup).Assembly.GetName().Name)
+                .Select(s => ((AssemblyPart)s).Assembly)
+                .ToList());
+
+            return this;
+        }
+
+        /// <summary>
+        /// Configures the application to use a constant culture for testing purposes.
+        /// </summary>
+        /// <returns>An instance of this <see cref="MvcWebApplicationBuilder{TStartup}"/></returns>
+        public MvcWebApplicationBuilder<TStartup> UseCultureReplacer()
+        {
+            ConfigureAfterStartup(s => s.TryAddEnumerable(ServiceDescriptor.Transient<IStartupFilter, CultureReplacerStartupFilter>()));
+            return this;
+        }
+
+        /// <summary>
+        /// Configures the application content root.
+        /// </summary>
+        /// <param name="solutionName">The glob pattern to use for finding the solution.</param>
+        /// <param name="solutionRelativePath">The relative path to the content root from the solution file.</param>
+        /// <returns>An instance of this <see cref="MvcWebApplicationBuilder{TStartup}"/></returns>
+        public MvcWebApplicationBuilder<TStartup> UseSolutionRelativeContentRoot(
+            string solutionRelativePath,
+            string solutionName = "*.sln")
+        {
+            var applicationBasePath = AppContext.BaseDirectory;
+
+            var directoryInfo = new DirectoryInfo(applicationBasePath);
+            do
+            {
+                var solutionPath = Directory.EnumerateFiles(directoryInfo.FullName, solutionName).FirstOrDefault();
+                if (solutionPath != null)
+                {
+                    ContentRoot = Path.GetFullPath(Path.Combine(directoryInfo.FullName, solutionRelativePath));
+                    return this;
+                }
+
+                directoryInfo = directoryInfo.Parent;
+            }
+            while (directoryInfo.Parent != null);
+
+            throw new Exception($"Solution root could not be located using application root {applicationBasePath}.");
+        }
+
+        public TestServer Build()
+        {
+            using (new CultureReplacer())
+            {
+                var builder = new WebHostBuilder()
+                    .UseContentRoot(ContentRoot)
+                    .ConfigureServices(InitializeServices);
+
+                builder.UseStartup<TestStartup<TStartup>>();
+
+                return new TestServer(builder);
```

Comment:
I would really prefer that this package doesn't have anything related xunit anything in it. I think it's just not needed, and if we stick xunit stuff in here users of other testing frameworks going to want it taken out.

Follow my reasoning for a set.
1. We have an msbuild targets file that copies the deps file
2. Make it an msbuild error if your 'app' project doesn't have a deps file
3. Then at runtime if you don't have a deps file the only possibility is that you're shadow copied. This should throw an exception and tell you to turn off shadow copying.